nilearn/datasets/tests/test_utils.py

```diff
@@ -275,37 +274,46 @@ def test_filter_columns():
 
 
 def test_uncompress():
-    # Create dummy file
-    fd, temp = mkstemp()
-    os.close(fd)
-
     # Create a zipfile
     dtemp = mkdtemp()
     ztemp = os.path.join(dtemp, 'test.zip')
     ftemp = 'test'
-    with contextlib.closing(zipfile.ZipFile(ztemp, 'w')) as testzip:
-        testzip.writestr(ftemp, 'test')
-    datasets.utils._uncompress_file(ztemp, verbose=0)
-    assert(os.path.exists(os.path.join(dtemp, temp)))
-    shutil.rmtree(dtemp)
-
-    dtemp = mkdtemp()
-    ztemp = os.path.join(dtemp, 'test.tar')
-    with contextlib.closing(tarfile.open(ztemp, 'w')) as tar:
-        tar.add(temp)
-    datasets.utils._uncompress_file(ztemp, verbose=0)
-    assert(os.path.exists(os.path.join(dtemp, temp)))
-    shutil.rmtree(dtemp)
-
-    dtemp = mkdtemp()
-    ztemp = os.path.join(dtemp, 'test.gz')
-    f = gzip.open(ztemp, 'wb')
-    f.close()
-    datasets.utils._uncompress_file(ztemp, verbose=0)
-    assert(os.path.exists(os.path.join(dtemp, temp)))
-    shutil.rmtree(dtemp)
-
-    os.remove(temp)
+    try:
+        with contextlib.closing(zipfile.ZipFile(ztemp, 'w')) as testzip:
+            testzip.writestr(ftemp, 'test')
+        datasets.utils._uncompress_file(ztemp, verbose=0)
+        assert (os.path.exists(os.path.join(dtemp, ftemp)))
+        shutil.rmtree(dtemp)
+
+        dtemp = mkdtemp()
+        ztemp = os.path.join(dtemp, 'test.tar')
+        ftemp = 'test'
+
+        # Create dummy file in the dtemp folder, so that the finally statement
+        # can easily remove it
+        fd, temp = mkstemp(dir=dtemp)
+        os.close(fd)
+        with contextlib.closing(tarfile.open(ztemp, 'w')) as tar:
+            tar.add(temp, arcname=ftemp)
+        datasets.utils._uncompress_file(ztemp, verbose=0)
+        assert (os.path.exists(os.path.join(dtemp, ftemp)))
```

Comment:
same comment here space after ```assert``` can be removed

Reply:
or remove parentheses. they might make this look like a function call but  `assert` is a statement, not a function

if you want to add a message you can't have them
```
# ok
assert os.path.exists(ftemp), "error message"

# this always passes (non-empty tuple bool value is True):
assert(os.path.exists(ftemp), "error message")
```

Reply:
I agree,  but @KamalakerDadi told me that because nilearn uses assert as a function in other examples, it would be better to stick with a function syntax

Reply:
that's not using assert as a function (it isn't one), it's just wrapping an expression in parentheses. it was a very minor comment and if this is what is done in nilearn no problem :)

Reply:
What is the conclusion ? Does space needs to be removed or not ?

Reply:
with a space makes more sense (and no parentheses unless we think python 4 will
make assert a function :) ). conclusion: it's not very important, whichever
option is fine by me